File: leetcode/python/0007_reverse_integer.py
class Solution:
  def reverse(self, x: int) -> int:
    negative = x < 0
    res = 0
    x = abs(x)

    while x > 0:
      res *= 10
      res += x % 10
      x //= 10
    
    if res < -1 * pow(2, 31) or res > pow(2, 31) - 1: return 0
    if negative: res *= -1
    return res

File: leetcode/cpp/0007_reverse_integer.cpp
class Solution {
public:
    int reverse(int x) {
        int reverseNum = 0;
        bool negative = false;

        if (x < 0) {
            negative = true;
            x = abs(x);
        }

        while (x != 0 && sizeof(reverseNum) == 4) {
            if ((10 * (long)reverseNum) + (x % 10) > pow(2, 31) - 1 || (10 * (long)reverseNum) + (x % 10) < -1 * pow(2, 31)) {
                return 0;
            }

            reverseNum = (10 * reverseNum) + (x % 10);
            x /= 10;
        }

        if (negative == true) {
            return reverseNum * -1;
        }

        return reverseNum;
    }
};